File: src/repetita/content/notetypes/phrase.py
"""
Something to say out loud, judged by the learner.

The one place where self-assessment is honest, because nothing else can hear it
yet.
"""

from __future__ import annotations

from ..models import CardTemplate, NoteType
from ._spec import field as f

phrase = NoteType(
    name="phrase",
    fields={
        "situation": f(required=True),
        "translation": f(),
        "target": f(required=True, visibility="after"),
        "explain": f(visibility="after"),
        "audio": f("audio", visibility="after"),
    },
    cards={
        "say": CardTemplate(
            ask=("situation", "translation"),
            expect="target",
            grader="self",
            # Flashcard only: `self` reads a rating out of the payload, and a
            # word bank submits assembled text, which it would score as AGAIN
            # every time. Declared here since this type was written and
            # unreachable behind `flashcard`, so nobody ever met it -- the
            # registry's own checks are what finally said so.
            forms=("flashcard",),
        ),
    },
)

File: latest/javascripts/repetita/types/phrase.js
// Something to say out loud, given a situation.
//
// The situation is a scene, not a question -- "you arrive at training and greet
// the group" -- so it reads as one, and the translation sits under it as the
// thing you are aiming to produce.

import { el } from "../dom.js";

export const notetype = "phrase";

export function question(card) {
  return el("div", { class: "question" }, [
    card.fields.situation ? el("p", { class: "scene", text: String(card.fields.situation) }) : null,
    card.fields.translation ? el("p", { class: "ask", text: String(card.fields.translation) }) : null,
  ]);
}
